static/app/components/compactSelect/types.tsx

```diff
@@ -0,0 +1,36 @@
+import {MenuListItemProps} from 'sentry/components/menuListItem';
+
+export interface SelectOption<Value extends React.Key> extends MenuListItemProps {
+  /**
+   * The option's value, must be unique within the selector.
+   */
+  value: Value;
+}
```

Comment:
This is similar to `SelectValue` in `types/core`, but it's a bit more constrained: the `Value` type must extend `React.Key` (`react-aria` only works with `Key`-type values), while `SelectValue` seems to allow any value type. We should consider consolidating the two, but I'm not sure if that's possible yet/any time soon.